packages/devextreme-react/src/core/widget-config.ts

```diff
@@ -1,11 +1,34 @@
 import { ITemplateMeta } from './template';
+import { ComponentBaseProps } from './component-base-func';
 
 const elementPropNames = ['style', 'id'];
 const classNamePropName = 'className';
 const refPropName = ['dropZone', 'dialogTrigger'];
 
+type InternalProps = {
+  [Property in keyof ComponentBaseProps]-?: ComponentBaseProps[Property];
+};
+
+const internalProps: InternalProps = {
+  WidgetClass: {},
+  isPortalComponent: false,
+  defaults: {},
+  templateProps: [],
+  expectedChildren: {},
+  subscribableOptions: [],
+  independentEvents: [],
+  useRequestAnimationFrameFlag: false,
+  clearExtensions: () => undefined,
+  renderChildren: () => undefined,
+  beforeCreateWidget: () => undefined,
+  afterCreateWidget: () => undefined,
+};
+
 function isIgnoredProp(name: string) {
-  return name === 'children' || name === classNamePropName || elementPropNames.includes(name);
+  return name === 'children'
+    || name === classNamePropName
+    || elementPropNames.includes(name)
+    || Object.prototype.hasOwnProperty.call(internalProps, name);
```

Comment:
In this file, we parse all props and those props that we don't filter out will (or may) be sent to the Widgets in `optionChanged` events. As now base components have many props for internal usage (that previously used to be inherited class properties), we need to filter them out to avoid spamming Widgets with invalid `optionChanged` when these props change.

To ensure that future us won't forget to exclude any new "internal" props, I created a typed object of the internal props type, but with all props marked as required, and filter out all its props at runtime.